src/pages/projects/[projectSlug]/index.js

```diff
@@ -0,0 +1,62 @@
+import { getNextStaticProps, is404 } from '@faustjs/next';
```

Comment:
`is404` is unused here, but that raises the question - how should we handle 404s? `is404` only works for the built-in post types, so maybe we emulate [what it's doing with `inlineResolved`](https://github.com/wpengine/faustjs/blob/be406fd048e097db6445ae793de5fc11efaed90e/packages/next/src/server/getProps.tsx#L130-L137) and the [subsequent checks](https://github.com/wpengine/faustjs/blob/be406fd048e097db6445ae793de5fc11efaed90e/packages/next/src/server/getProps.tsx#L195-L199)? @blakewilson

Reply:
Instead of re-creating the `is404` logic, we could go the middleware route. This could open the door to a future middleware that checks if a uri exists on WP, instead of relying on `is404`, which only covers a subset of posts:

We could put something like this in `src/pages/projects/[projectSlug]/_middleware.js`
```js
import { NextResponse } from 'next/server';
export default async function _middleware(req, ctx) {
  const { origin } = new URL(req.url);
  const wpUrl = process.env.NEXT_PUBLIC_WORDPRESS_URL;
  const projectSlug = req?.page?.params?.projectSlug;

  const is404 = await fetch(`${wpUrl}/graphql`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      query: `query ProjectQuery($projectSlug:ID!) {
            project(id: $projectSlug, idType: SLUG) {
                id
            }
        }`,
      variables: {
        projectSlug,
      },
    }),
  });

  const { data } = await is404.json();

  if (!data?.project?.id) {
    return NextResponse.rewrite(`${origin}/404`);
  }

  return NextResponse.next();
}
```

Reply:
This is a potential performance bottleneck as the site might get many 404 requests (from bots or whatever)

Reply:
I agree we should make this easier in the future, maybe with middleware or maybe with and update to `is404()`. It sounds like those approaches might need more discussion though. Should we just stick with a manual approach that uses `inlineResolved` in `getStaticProps` for now?